File: DS using c/queues/queues-arr.c
#include <stdio.h>
#define MAX_SIZE 10

int front = -1, rear = -1;
int queue[MAX_SIZE];
void enqueue()
{
    if(rear == -1 && front == -1)
            front=0, rear = 0;
    else if(rear == MAX_SIZE-1){
        printf("Queue is full");
        return;
    }
    else rear += 1;
    printf("\nEnter value : ");
    scanf("%d",&queue[rear]);
    prtinf("\nValue added successfully\n");
}

void dequeue()
{
    if(rear == -1 && front == -1)
            printf("Queue is empty");
    else if(front==rear){
        printf("%d is removed",queue[front]);
        front=-1,rear=-1;
    }
    else{
        front+=1;
        printf("%d is removed",queue[front-1]);
    }
}

void display()
{
    if (rear == -1 && front == -1)
        printf("\nQueue is empty");
    else
    {
        printf("\nQueue elements are : ");
        for (int i = front; i <= rear; i++)
            printf("%d ", queue[i]);
        printf("\n");
    }
}

void perform_operation(int option)
{
    switch (option)
    {
    case 1:
        enqueue();
        break;
    case 2:
        dequeue();
        break;
    case 3:
        display();
        break;
    case 4:
        printf("\n----------Program ends----------\n");
        break;
    default:
        printf("\nSelect a valid option\n");
    }
}

void start_operation()
{
    int option;
    do
    {
        printf("\n1. Enqueue an Element on to Queue"
               "\n2. Dequeue an Element from Queue"
               "\n3. Display the elements of a Queue"
               "\n4. Exit\n");
        printf("\nSelect a option from above: ");
        scanf("%d", &option);
        perform_operation(option);
    } while (option != 4);
}

int main()
{
    printf("\n-----------Implementation of Queue using array-----------\n\n");
    start_operation();
    return 0;
}

File: DS using python/queues/queues-arr.py
class Queue:
    def __init__(self):
        self.front = -1
        self.rear = -1
        self.size = 10
        self.queue = list(range(self.size))

    def enqueue(self):
        if self.rear == -1 and self.front == -1:
            self.front = self.rear = 0
        elif self.rear == self.size-1:
            print("Queue is full")
            return
        else:
            self.rear += 1
        self.queue[self.rear] = int(input("Enter value: "))

    def dequeue(self):
        if self.rear == -1 and self.front == -1:
            print("Queue is empty")
        elif self.front==self.rear:
            self.front=self.rear=-1
        else:
            self.front+=1
            print(self.queue[self.front-1],"is removed")

    def display(self):
        if self.rear == -1 and self.front == -1:
            print("Queue is empty")
        else:
            print("Queue elements are : ",end=' ')
            for i in range(self.rear,self.front-1,-1):
                print(self.queue[i],end=" ")
            print()
    
    def perform(self,option):
        if option not in range(5):
            print("Enter a valid option\n")
        else:
            if option==1:
                self.enqueue()
            elif option==2:
                self.dequeue()
            elif option==3:
                self.display()
            elif option==4:
                print("====================program ends====================")

    def start(self):
        while(True):
            print("1. Enqueue item\n"+
                  "2. dequeue item\n"+
                  "3. display item\n"+
                  "4. exit\n")
            option=int(input("Enter option: "))
            self.perform(option)
            if(option==4): break        

if __name__ == "__main__":
    q=Queue()
    print("++++++++++++++++++Implementation of Queue using arrays++++++++++++++++++")
    q.start()   
